src/cmake/swig_lib.cmake

```diff
@@ -62,16 +62,29 @@ function(swig_lib)
                                         -Werror
                                         -w317,325,378,401,402,467,472,503,509)
 
-  set_property(SOURCE ${ARG_I_FILE}
-               PROPERTY SWIG_MODULE_NAME ${ARG_NAME})
+  if (DEFINED ARG_MODULE_NAME)
+    set_property(SOURCE ${ARG_I_FILE}
+                 PROPERTY SWIG_MODULE_NAME ${ARG_MODULE_NAME})
+  else()
+    set_property(SOURCE ${ARG_I_FILE}
+                 PROPERTY SWIG_MODULE_NAME ${ARG_NAME})
+  endif()
+  
   set_property(SOURCE ${ARG_I_FILE}
                PROPERTY USE_SWIG_DEPENDENCIES TRUE)
   set_property(SOURCE ${ARG_I_FILE}
                PROPERTY USE_TARGET_INCLUDE_DIRECTORIES true)
 
+
+  if (ARG_SHARED_LIBRARY)
```

Comment:
https://cmake.org/cmake/help/latest/module/UseSWIG.html seems to have a `USE_BUILD_SHARED_LIBS` option to achieve something similar.